xcat3/common/rest.py

```diff
@@ -0,0 +1,65 @@
+from oslo_log import log as logging
+from xcat3.common import exception
+
+import requests
+import json
+
+LOG = logging.getLogger(__name__)
+
+class RestSession :
+
+    headers = {'Content-Type': 'application/json'}
+
+    def __init__(self) :
+        self.session = requests.Session()
+
+    def _request_log(self, method, url, data) :
+        log_string = ['curl -k']
+        log_string.append(' -X %s' % method)
+        log_string.append(' -H %s' % RestSession.headers)
+        log_string.append(" '%s'" % url)
+
+        if data :
+            log_string.append('-d %s' % data)
+
+        LOG.info("REQ: %s" % "".join(log_string))
+
+    def _rspcheck(self, rsp) :
+        if rsp.text and rsp.status_code != 400 :
+            try :
+                body = json.loads(rsp.text)
+            except ValueError :
+                body = None
+        else :
+            body = None
+
+        LOG.info("RESP: [%(status)s]\nRESP BODY: %(text)s\n",
+                {'status': rsp.status_code, 'text': json.dumps(body)})
+
+        if rsp.status_code >= 400 :
+            rsp_data = rsp.json()
+            raise exception.ExceptionFromRestRsp(exception=rsp_data['data']['description'])
+
+    def request (self, method, url, in_data) :
+        if in_data :
+            data = json.dumps(in_data)
+        else :
+            data = ''
+
+        self._request_log(method, url, data)
+
+        try :
+            if method == 'POST' :
```

Comment:
Does the methods like POST PUB GET PUT can be replaced by `session.request(method,url,**kwargs)` ? As a common library, you should not specify the headers simply like `headers = {'Content-Type': 'application/json'}`

You'd better split the line into multiple lines since it beyond 79 characters.